packages/engine/Source/Renderer/TextureAtlas.js

```diff
@@ -671,54 +679,76 @@ TextureAtlas.prototype.addImage = function (id, image) {
       return -1;
     }
 
-    return this._addImage(index, image);
+    return this._addImage(index, image).then((index) => {
+      this._indexPromiseById.delete(id);
```

Comment:
This change is necessary in order to realize in `addImageSubRegion` that the image has already been added and resolved.

I made sure to account for this change in the relevant places

Reply:
It shouldn't be possible, all promises created on the current tick will resolve at best on the next tick, and on this tick the promise will safely be set. (not really ticks its the microtask queue but its the same answer regardless)

Reply:
I'll move it out anyways though I think its more readable that we syntactically run `set` before `delete`